### DISCUSSION TITLE
Set PRAGMA when using SQlite3

### DISCUSSION THREAD

**Discussion by solracsf** [juicedata/juicefs#2615] [Ideas]
When using Litestream, there are some parameters that can be optimized : https://litestream.io/tips/
Is there any way to pass PRAGMA parameters? Could this be an enhancement?

This would avoid lines like:

```
juicefs[634272] <WARNING>: Transaction succeeded after 3 tries (3.671605ms), inodes: [1], last error: database is locked [sql.go:654]
```

As DB is certainly locked by Litestream when snapshotting.

**Comment by davies** [comment]
Are you using Litestream in production?

Maybe we can pass the tuning parameter using URL query (into meta-url), and support them one by one.

**Reply by solracsf** [reply]
> Are you using Litestream in production?

Yes, as a backup/snapshot solution.
